src/main/java/org/openmrs/addonindex/domain/AddOnInfoAndVersions.java

```diff
@@ -24,152 +25,169 @@
+
+    public int getVersionCount() {
+        return versions == null ? 0 : versions.size();
+    }
+
+    public Version getLatestVersion() {
+        return versions == null || versions.size() == 0 ? null : versions.get(0).getVersion();
+    }
+
+    public String getUid() {
+        return uid;
+    }
+
+    public void setUid(String uid) {
+        this.uid = uid;
+    }
+
+    public AddOnStatus getStatus() {
+        return status;
+    }
+
+    public void setStatus(AddOnStatus status) {
+        this.status = status;
+    }
+
+    public AddOnType getType() {
+        return type;
+    }
+
+    public void setType(AddOnType type) {
+        this.type = type;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getIcon() {
+        return icon;
+    }
+
+    public void setIcon(String icon) {
+        this.icon = icon;
+    }
+
+    public List<String> getTags() {
+        return tags;
+    }
+
+    public void setTags(List<String> tags) {
+        this.tags = tags;
+    }
+
+    public List<Maintainer> getMaintainers() {
+        return maintainers;
+    }
+
+    public void setMaintainers(List<Maintainer> maintainers) {
+        this.maintainers = maintainers;
+    }
+
+    public String getHostedUrl() {
+        return hostedUrl;
+    }
+
+    public void setHostedUrl(String hostedUrl) {
+        this.hostedUrl = hostedUrl;
+    }
+
+    public List<AddOnVersion> getVersions() {
+        return versions;
+    }
+
+    public void setVersions(List<AddOnVersion> versions) {
+        this.versions = versions;
+    }
+
+    public Optional<AddOnVersion> getVersion(Version version) {
+        if (versions == null) {
+            return null;
+        }
+        return versions.stream().filter(v -> v.getVersion().equals(version)).findFirst();
+    }
+
+    public void addTag(String tag) {
+        if (!Pattern.matches("[^\\s]*", tag)) {
+            throw new IllegalArgumentException("Tag cannot contain whitespace:" + tag);
+        }
+        if (tags == null) {
+            tags = new ArrayList<>();
+        }
+        tags.add(tag);
+    }
+
+    public AddOnVersion getLatestSupportedVersion(String userCoreVersion) throws Exception {
+        if (userCoreVersion != null) {
+            if (userCoreVersion.equals("Default")){
+                return getVersion(getLatestVersion()).get();
+            }
+            else {
+                for (AddOnVersion addOnVersion : getVersions()) {
+                    if (OpenmrsVersionCompareUtil.matchRequiredVersions(userCoreVersion, addOnVersion.getRequireOpenmrsVersion())) {
+                        return addOnVersion;
+                    }
+                }
+                return new AddOnVersion();
```

Comment:
I guess that in the controller you'd have to return a different HTTP Status code on null (example: https://stackoverflow.com/questions/32396884/return-http-204-on-null-with-spring-restcontroller)

And then in the react code that calls this controller you'd need to check the HTTP status before constructing any object.